src/mappings/entities/mapIdentities.ts

```diff
@@ -0,0 +1,192 @@
+        transactionGroups: [],
+        createdBlockId: blockId,
+        datetime,
+      }).save();
+      await Account.create({
+        id: address,
+        identityId: did,
+        permissionsId: address,
+        eventId,
+        address,
+        blockId,
+        datetime,
+      }).save();
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysAdded) {
+      const { id: identityId } = await fetchIdentity(params[0]);
+
+      const accountsList = params[1].toJSON() as any[];
+
+      const promises = [];
+
+      for (const account of accountsList) {
+        const address = account.signer.account;
+
+        const { assets, portfolios, transactions, transactionGroups } = getPermissions(
+          account.permissions
+        );
+
+        promises.push(
+          Permissions.create({
+            id: address,
+            assets,
+            portfolios,
+            transactions,
+            transactionGroups,
+            createdBlockId: blockId,
+            datetime,
+          }).save()
+        );
+
+        promises.push(
+          Account.create({
+            id: address,
+            address,
+            identityId,
+            permissionsId: address,
+            eventId,
+            blockId,
+            datetime,
+          }).save()
+        );
+      }
+
+      await Promise.all(promises);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysFrozen) {
+      const identity = await fetchIdentity(params[0]);
+      identity.secondaryKeysFrozen = true;
+      await identity.save();
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysUnfrozen) {
+      const identity = await fetchIdentity(params[0]);
+      identity.secondaryKeysFrozen = false;
+      await identity.save();
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeysRemoved) {
+      await fetchIdentity(params[0]);
+      const accountsRemoved = params[1].toJSON() as any[];
+      const accounts = accountsRemoved.map(account => Account.remove(account.account));
+
+      await Promise.all(accounts);
+    }
+
+    if (eventId === EventIdEnum.SecondaryKeyPermissionsUpdated) {
+      await fetchIdentity(params[0]);
+      const address = getFirstValueFromJson(params[1])['account'];
+      const permissions = await Permissions.get(address);
+      const { assets, portfolios, transactions, transactionGroups } = getPermissions(
+        params[3].toJSON() as any
+      );
+
+      permissions.assets = assets;
+      permissions.portfolios = portfolios;
+      permissions.transactions = transactions;
+      permissions.transactionGroups = transactionGroups;
+      permissions.updatedBlockId = `${blockId}`;
+      await permissions.save();
+    }
+  }
+}
+
+function getPermissions(accountPermissions: Record<string, unknown>) {
+  logger.info(JSON.stringify(accountPermissions));
+  let assets,
+    portfolios,
+    transactions,
+    transactionGroups = [];
+
+  let type;
+  Object.keys(accountPermissions).forEach(key => {
+    logger.info(key);
```

Comment:
Should be removed. guesing this is left over from debugging.

If we are logging stuff thats not for debugging it should have more context.